static/js/components/EducationTab.js

```diff
@@ -0,0 +1,249 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Button from 'react-mdl/lib/Button';
+import ProfileTab from "../util/ProfileTab";
+import { generateNewEducation } from "../util/util";
+import _ from 'lodash';
+import {
+  boundMonthField,
+  boundYearField,
+  saveAndContinue
+} from '../util/profile_edit';
+import {
+  toggleEducationLevel,
+  openEducationForm,
+  closeEducationForm
+} from '../actions';
+import { HIGH_SCHOOL, ASSOCIATE, BACHELORS, MASTERS, DOCTORATE } from '../constants';
+import { Grid, Cell, Switch, Dialog, DialogTitle, DialogContent, DialogActions, FABButton, Icon } from 'react-mdl';
+
+class EducationTab extends ProfileTab {
+  constructor(props) {
+    super(props);
+    this.boundMonthField = boundMonthField.bind(this);
+    this.boundYearField = boundYearField.bind(this);
+    this.educationLevelOptions = [
+      {value: HIGH_SCHOOL, label: "High school"},
+      {value: ASSOCIATE, label: 'Associate degree'},
+      {value: BACHELORS, label: "Bachelor's degree"},
+      {value: MASTERS, label: "Master's or professional degree"},
+      {value: DOCTORATE, label: "Doctorate"}
+    ];
+
+    this.educationLevelLabels = {};
+    this.educationLevelOptions.forEach(level => {
+      this.educationLevelLabels[level.value] = level.label;
+    });
+    this.saveAndContinue = saveAndContinue.bind(this, '/profile/professional', EducationTab.validation);
+    this.openNewEducationForm = this.openNewEducationForm.bind(this);
+    this.handleCloseDialog = this.handleCloseDialog.bind(this);
+
+  }
+  static propTypes ={
+    educationLevels: React.PropTypes.object,
+    educationDialog: React.PropTypes.object,
+    profile: React.PropTypes.object,
+    saveProfile: React.PropTypes.func,
+    updateProfile: React.PropTypes.func,
+    errors: React.PropTypes.object,
+    dispatch: React.PropTypes.func.isRequired
+  };
+
+  static defaultProps = {
+    requiredFields: [],
+    validationMessages: {
+      'degree_name': "'Degree level",
+      'graduation_date': 'Graduation Date',
+      'field_of_study': 'Field of study',
+      'online_degree': 'Online Degree',
+      'school_name': 'School name',
+      'school_city': 'City',
+      'school_state_or_territory': 'State',
+      'school_country': 'Country'
+    }
+  };
+
+  static validation(profile, requiredFields) {
+    const keys = [
```

Comment:
can you refactor this so that the keys for validating nested objects are a class property of the `EducationTab` class?

should be pretty simple, something like

```js
static nestedValidationFields = [ ... ];

validation ( ... ) {
  let nestedFields = (index) => {
    let keySet = (key) => ['education', index, key];
    return EducationTab.nestedValidationKeys.map(key => keySet(key));
  };
}
```

this will make it easier to implement the complete profile check later for the education tab (specifically, it's handy for testing).